features/page-objects/RoomSettingPage.js

```diff
@@ -36,12 +37,12 @@ class RoomSettingPage extends Page {
 
   async setAccessCode(accessCode) {
     const accessCodeInput = await this.findByCss("[id='room_access_code']");
```

Comment:
I assume all these async functions are controlling the same page: `RoomSettingPage`. So why is the`enterAccessCode` function and `setAccessCode` function setting `accessCodeInput` using 2 different ids?

`id='waiting_room_access_code'` and  `id='room_access_code'`. Are they interacting with the same field and that field's id selector changes based on the current state of the page? Or are these 2 different fields?

Reply:
It _looks like_ we probably are conflating two distinct pages; the "WaitingRoom" page, which someone who is not allowed into the room without entering the access code; and the "RoomSettingsPage" which is where a Space owner _sets_ the access code for the space.

May be worth splitting those out!

Reply:
Agreed. The "WaitingRoom" page and "RoomSettingsPage" might display the same elements right now, but if the elements have different ids, then they should have separate page classes. Else it'll get confusing. Also if the pages need to execute the same functions, those functions can be stored in a separate file to be DRY.

The "RoomSettingsPage" may also change in the future when extra settings options are added.